### PR TITLE
Make the banner image change like the course tile does

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**
Currently you can trigger this by having a my-courses widget on your banner page and changing the course's image via that

The file was getting huge, so I also broke the styles into their own file

### CODE REVIEW

**Comment by jstefaniuk-d2l** [comment]
So this can either be code reviewed and tested, or I can add the image selector in before code review/test

File: src/d2l-image-banner-overlay.html [hunk 1]
```diff
@@ -282,6 +115,75 @@ <h1 id="courseName" class="d2l-heading-1">[[_courseName]]</h1>
...
+
+				var orgSelfLink = (org.getLinkByRel('self') || {}).href || '',
+					urlPosition = orgSelfLink.indexOf(this.organizationUrl);
+
+				if (urlPosition === (orgSelfLink.length - this.organizationUrl.length)) {
+					switch (e.detail.status) {
+						case 'set':
+							if (e.detail.image) {
+								var newImageHref = this.getDefaultImageLink(e.detail.image);//,
+								//	newSrcset = this.getImageSrcset(e.detail.image);
```
Comment (cgalvind2l) [inline src/d2l-image-banner-overlay.html]: Why is this code commented out?

File: src/d2l-image-banner-overlay.html [hunk 2]
```diff
@@ -282,6 +115,75 @@ <h1 id="courseName" class="d2l-heading-1">[[_courseName]]</h1>
...
+
+								var overlayContent = this.$.overlayContent;
+								this.toggleClass('change-image-loading', true, overlayContent);
+								// load the image while the loading spinner runs to that we have it when the spinner ends
+								// NOTE: if this needs optimization, we can wait for the image's onload to play the success animation
+								this._nextImage = e.detail.image;
+								var imagePreloader = document.createElement('img');
+								imagePreloader.setAttribute('src', newImageHref);
+								// imagePreloader.setAttribute('srcset', newSrcset);
```
Comment (cgalvind2l) [inline src/d2l-image-banner-overlay.html]: Same as above, why are these 2 lines of code commented out?

**Comment by CodeBaboon** [comment]
Code reviewed offline (I had some questions before I could make suggestions). Ultimately the suggested improvements are:

- Refactor the code that determines the image src for both the preloader image and the actual banner display image to use a common function so that we are guaranteed to always use the same src url.(otherwise the preloader image might preload some other file and just be a waste of time/resources).
- Remove the commented out lines of code.

I will do testing once these suggestions are implemented.

File: bower.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
     "d2l-alert": "git://github.com/Brightspace/alert.git#^0.0.3",
     "d2l-colors": "^2.2.3",
     "d2l-dropdown": "^5.0.0",
+    "d2l-organization-hm-behavior": "git://github.com/Brightspace/organization-hm-behavior.git#~0.0.1",
```
Comment (cgalvind2l) [inline bower.json]: Nit: maybe this could be put in the correct spot alphabetically?

**Comment by CodeBaboon** [comment]
Tested the following scenarios across Chrome, FF, Edge, and IE:

- [x] When My Courses widget is on homepage, select "Change Image" for current course, pick a new image, banner transitions to spinner then to checkmark and new image loads.
- [x] When My Courses widget is on homepage, select "Change Image" and cancel the dialog. Nothing changes.
- [x] When My Courses widget is on homepage, select "Change Image" for current course, turn off internet pick a new image, banner transitions to spinner and then failure mark as expected.
- [x] When My Courses widget is on homepage, select "Change Image" for a different course, pick a new image, my courses widget updates appropriately but current course banner correctly remains unchanged.

**Pull request merged**